tests/test_zuora_discovery.py

```diff
@@ -0,0 +1,129 @@
+from tap_tester import menagerie, connections
+from tap_tester.logger import LOGGER
+
+from base import ZuoraBaseTest
+
+class DiscoveryTest(ZuoraBaseTest):
+    """Test tap discovery mode and metadata/annotated-schema conforms to standards."""
+
+    @staticmethod
+    def name():
+        return "tap_tester_zuora_discovery_rest"
+
+    def test_run(self):
+        """ Executing tap-tester scenarios for both types of zuora APIs AQUA and REST"""
+        self.run_test("AQUA")
+        self.run_test("REST")
+
+    def run_test(self, api_type):
+        """
+        Verify that discover creates the appropriate catalog, schema, metadata, etc.
+
+        • Verify number of actual streams discovered match expected
+        • Verify the stream names discovered were what we expect
+        • Verify stream names follow naming convention
+          streams should only have lowercase alphas and underscores
+        • verify there is only 1 top level breadcrumb
+        • Verify primary key(s) match expectations.
+        • Verify replication key(s) match expectations.
+        • verify that if there is a replication key we are doing INCREMENTAL otherwise FULL
+        • verify the actual replication matches our expected replication method
+        • verify that primary, replication and foreign keys
+          are given the inclusion of automatic (metadata and annotated schema).
+        • verify that all other fields have inclusion of available (metadata and schema)
+        """
+        self.zuora_api_type = api_type
+
+        streams_to_test = self.expected_streams()
+
+        conn_id = connections.ensure_connection(self, original_properties=False)
+
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # verify the tap only discovers the expected streams
+        found_catalog_names = {catalog['tap_stream_id'] for catalog in found_catalogs}
+        self.assertSetEqual(streams_to_test, found_catalog_names)
+        LOGGER.info("discovered schemas are OK")
+
+        for stream in streams_to_test:
+            with self.subTest(stream=stream):
+                catalog = next(iter([catalog for catalog in found_catalogs
+                                     if catalog["stream_name"] == stream]))
+                assert catalog  # based on previous tests this should always be found
```

Comment:
Use unittest assertion `self.assertIsNotNone()`.

Reply:
made the changes